app/models/manageiq/providers/azure/cloud_manager/provision/state_machine.rb

```diff
@@ -0,0 +1,9 @@
+module ManageIQ::Providers::Azure::CloudManager::Provision::StateMachine
+  def customize_destination
```

Comment:
@durandom - Like you, my instinct was to delete the IPs during instance retirement but then I realized it is standard practice across cloud providers not to in favor of reusing IP addresses through floating IP assignment.